hibernate-core/src/main/java/org/hibernate/query/sqm/spi/NamedSqmQueryMemento.java

```diff
@@ -47,68 +41,68 @@
 	Map<String, String> getParameterTypes();
 
 	@Override
-	NamedHqlQueryMemento makeCopy(String name);
+	NamedSqmQueryMemento makeCopy(String name);
 
 	/**
 	 * Delegate used in creating named HQL query mementos.
 	 *
 	 * @see NamedHqlQueryDefinition
 	 * @see NameableQuery#toMemento
 	 */
-	class Builder extends AbstractNamedQueryMemento.AbstractBuilder<Builder> {
-		protected String hqlString;
+	class Builder extends AbstractNamedQueryMemento.AbstractBuilder<NamedSqmQueryMemento.Builder> {
+		protected SqmStatement sqmSelectStatement;
 		protected LockOptions lockOptions;
 		protected Integer firstResult;
 		protected Integer maxResults;
-		protected Map<String,String> parameterTypes;
+		protected Map<String, String> parameterTypes;
 
 
 		public Builder(String name) {
 			super( name );
 		}
 
 		@Override
-		protected Builder getThis() {
+		protected NamedSqmQueryMemento.Builder getThis() {
 			return this;
 		}
 
-		public Builder setReadOnly(boolean readOnly) {
+		public NamedSqmQueryMemento.Builder setReadOnly(boolean readOnly) {
 			this.readOnly = readOnly;
 			return this;
 		}
 
-		public Builder addParameterType(String name, String typeName) {
+		public NamedSqmQueryMemento.Builder addParameterType(String name, String typeName) {
 			if ( this.parameterTypes == null ) {
 				this.parameterTypes = new HashMap<>();
 			}
 			this.parameterTypes.put( name, typeName );
 			return this;
 		}
 
-		public Builder setParameterTypes(Map<String,String> parameterTypes) {
+		public NamedSqmQueryMemento.Builder setParameterTypes(Map<String, String> parameterTypes) {
 			this.parameterTypes = parameterTypes;
 			return this;
 		}
 
-		public Builder setLockOptions(LockOptions lockOptions) {
+		public NamedSqmQueryMemento.Builder setLockOptions(LockOptions lockOptions) {
 			this.lockOptions = lockOptions;
 			return this;
 		}
 
-		public Builder setFirstResult(Integer firstResult) {
+		public NamedSqmQueryMemento.Builder setFirstResult(Integer firstResult) {
 			this.firstResult = firstResult;
 			return this;
 		}
 
-		public Builder setMaxResults(Integer maxResults) {
+		public NamedSqmQueryMemento.Builder setMaxResults(Integer maxResults) {
 			this.maxResults = maxResults;
 			return this;
 		}
 
-		public NamedHqlQueryMemento createNamedQueryDefinition() {
-			return new NamedHqlQueryMementoImpl(
+		public NamedSqmQueryMemento createNamedQueryDefinition() {
+			return new NamedCriteriaQueryMementoImpl(
```

Comment:
Shouldn't this return either a `NamedHqlQueryMementoImpl` or `NamedCriteriaQueryMementoImpl` depending on whether the `sqmSelectStatement` is set?

Reply:
this Builder is never used I'm going to remove it